genePred/compare.go

```diff
@@ -1,61 +1,68 @@
 package genePred
 
 import (
-	"log"
 	"strings"
 )
 
-func AllAreEqual(a []*GenePred, b []*GenePred) bool {
+// AllAreEqual examines if two genePred files are the same entry by entry.
+func AllAreEqual(a []GenePred, b []GenePred) (bool, []int) {
+	var fields []int
+	var equal bool
+
 	if len(a) != len(b) {
-		return false
+		return false, fields
 	}
 	for i := 0; i < len(a); i++ {
-		if !Equal(a[i], b[i]) {
-			return false
-		}
+		equal, fields = Equal(a[i], b[i])
+		return equal, fields
```

Comment:
I think you want to keep this return statement wrapped in an "if" check, otherwise you are only going to be returns the answer from the first time through the loop (i.e. a[0] vs b[0]).